packages/ipfs-cli/src/daemon.js

```diff
@@ -27,7 +27,13 @@ class Daemon {
     }
   }
 
-  async start () {
+  /**
+   * Starts the IPFS HTTP server
+   * @param {object} [opts] - specify advanced configuration
+   * @param {object} [opts.ipld.formats] - IPLD custom formats
+   * @return {Promise<HttpApi>}
+   */
+  async start (opts = {}) {
```

Comment:
How are the opts passed into daemon start? If I run `jsipfs daemon` it's not going to pass anything into the start function, so there's no way to set these.

It would be helpful to have a test showing how passing the formats is intended to work, it's not clear to me right now, other than directly using the http server and client.

The updates themselves look good but the usage isn't clear.

Reply:
You can't do this through the CLI as it stands, instead you'd create a wrapper that starts a daemon with the required IPLD options.

I've added an example of how to do this in #3347